tests/test_wa_monthly_views_common.py

```diff
@@ -0,0 +1,384 @@
+                            observations = [Obs(variable=var_precip_net1_1, history=history_stn1_hourly,
+                                                time=datetime.datetime(2000, 1, day, hour), datum=1.0)
+                                            for day in days for hour in hours]
+                        for sesh in generic_sesh(variable_sesh , observations):
+                            yield sesh
+
+                    @fixture
+                    def variable(request, var_temp_point, var_precip_net1_1):
+                        """Yield a Variable object (which is a fixture), dependent on the value of request.param:
+                         MonthlyAverageOfDailyMaxTemperature, MonthlyAverageOfDailyMinTemperature, or MonthlyTotalPrecipitation.
+                        The variable yielded is the one expected for the type of view/test indicated by request.param.
+                        This fixture is used as an indirect fixture for parametrized tests.
+                        """
+                        variable = {
+                            MonthlyAverageOfDailyMaxTemperature: var_temp_point,
+                            MonthlyAverageOfDailyMinTemperature: var_temp_point,
+                            MonthlyTotalPrecipitation: var_precip_net1_1,
+                        }[request.param]
+                        yield variable
+
+                    @mark.parametrize('View, obs_sesh', [
+                        (MonthlyAverageOfDailyMaxTemperature, MonthlyAverageOfDailyMaxTemperature),
+                        (MonthlyAverageOfDailyMinTemperature, MonthlyAverageOfDailyMinTemperature),
+                        (MonthlyTotalPrecipitation, MonthlyTotalPrecipitation),
+                    ], indirect=['obs_sesh'], ids=id)
+                    def it_returns_a_single_row(View, obs_sesh):
+                        refresh_views(obs_sesh)
+                        assert obs_sesh.query(View).count() == 1
+
+                    @mark.parametrize('View, obs_sesh, variable', [
+                        (MonthlyAverageOfDailyMaxTemperature, MonthlyAverageOfDailyMaxTemperature, MonthlyAverageOfDailyMaxTemperature),
+                        (MonthlyAverageOfDailyMinTemperature, MonthlyAverageOfDailyMinTemperature, MonthlyAverageOfDailyMinTemperature),
+                        (MonthlyTotalPrecipitation, MonthlyTotalPrecipitation, MonthlyTotalPrecipitation),
+                    ], indirect=['obs_sesh', 'variable'], ids=id)
+                    def it_returns_the_expected_history_variable_and_day(View, obs_sesh, history_stn1_hourly, variable):
+                        refresh_views(obs_sesh)
+                        result = obs_sesh.query(View).first()
+                        assert result.history_id == history_stn1_hourly.id
+                        assert result.vars_id == variable.id
+                        assert result.obs_month == datetime.datetime(2000, 1, 1)
+
+                    @mark.parametrize('View, obs_sesh, statistic', [
```

Comment:
Pytest has some arcane features. That business of indirection is handy but hacky. And there are still others involving test collection and metafunc that I don't really grok yet. Doubtless I will look on this work as naive in a few months ...